hyco-https/tests/largepost.test.js

```diff
@@ -0,0 +1,90 @@
+var https = require('..')
+
+// create a large message over 64kb
+var largeMessage = "a";
+for (var i = 0; i < 17; i++) {
+    largeMessage += largeMessage;
+}
+
+var totalRequests = 10; // Total requests to send over the test
+jest.setTimeout(5000 + (totalRequests * 200)); // Expect 5 seconds + 5 requests per second
+
+test('HTTP POST', (done) => {
+    var ns = process.env.SB_HC_NAMESPACE ? process.env.SB_HC_NAMESPACE.replace(/^"(.*)"$/, '$1') : null;
+    var path = process.env.SB_HC_PATH ? process.env.SB_HC_PATH : "a2";
+    var keyrule = process.env.SB_HC_KEYRULE ? process.env.SB_HC_KEYRULE.replace(/^"(.*)"$/, '$1') : null;
+    var key = process.env.SB_HC_KEY ? process.env.SB_HC_KEY.replace(/^"(.*)"$/, '$1') : null;
+
+    expect(ns).toBeDefined();
+    expect(path).toBeDefined();
+    expect(keyrule).toBeDefined();
+    expect(key).toBeDefined();
+
+    var listenerCount = 0;
+    var senderCount = 0;
+
+    /* set up the listener */
+    var uri = https.createRelayListenUri(ns, path);
+    var server = https.createRelayedServer({
+            server: uri,
+            token: () => https.createRelayToken(uri, keyrule, key)
+        },
+        (req, res) => {
+            expect(req.method).toBe("POST");
+            expect(req.headers.custom).toBe("Hello");
+            req.setEncoding('utf-8');
+            req.on('data', (chunk) => {
+                expect(chunk).toBe(largeMessage);
+            });
```

Comment:
Should this test handle chunked request data like chunkedpost.test.js?

``` JS
    var chunks = '';
    req.setEncoding('utf-8');
    req.on('data', (chunk) => {
        chunks = chunks + chunk;
    });
    req.on('end', () => {
        expect(chunks).toBe('12345678901234567890Hello');
        res.write('Hello');
        res.end();
    });
```

Reply:
If you change the post body to be 1Mb or larger does this data come in chunks? (Maybe ours or the websocket library is already doing that, I can't recall).

---
In reply to: [316299342](https://github.com/Azure/azure-relay-node/pull/33#discussion_r316299342) [](ancestors = 316299342)